sympy/functions/combinatorial/factorials.py

```diff
@@ -849,10 +890,83 @@ def eval(cls, n, k):
             from sympy import gamma
             return gamma(n + 1)/(gamma(k + 1)*gamma(n - k + 1))
 
+    def _eval_Mod(self, q):
+        n, k = self.args
+        if all(x.is_integer for x in (n, k, q)):
```

Comment:
I think that

       if not all(x.is_integer for x in (n, k, q)):
           raise ValueError("Integers expected for binomial Mod")

would be more readable (and avoid one indentation step in the sequel).

Reply:
Yes, valid point.